x/posts/internal/types/post.go

```diff
@@ -334,8 +296,8 @@ type Posts []Post
 func (p Posts) String() string {
 	out := "ID - [Creator] Message\n"
 	for _, post := range p {
-		out += fmt.Sprintf("%d - [%s] %s\n",
-			post.PostID, post.Creator, post.Message)
+		out += fmt.Sprintf("%s - [%s] %s\n",
+			post.PostID.String(), post.Creator, post.Message)
```

Comment:
Since `PostID` implements the `String()` method, we can remove the call here as it will be implicit

```suggestion
			post.PostID, post.Creator, post.Message)
```